src/makernote_int.cpp

```diff
@@ -654,6 +654,13 @@ TiffComponent* newIfdMn2(uint16_t tag, IfdId group, IfdId mnGroup) {
 
 TiffComponent* newOlympusMn(uint16_t tag, IfdId group, IfdId /*mnGroup*/, const byte* pData, size_t size,
                             ByteOrder /*byteOrder*/) {
+  // FIXME: workaround for overwritten OM System header in Olympus files (https://github.com/Exiv2/exiv2/issues/2542)
+  if (size >= 14 && std::string(reinterpret_cast<const char*>(pData), 14) == std::string("OM SYSTEM\0\0\0II", 14)) {
```

Comment:
No need for the second std::string constructor.

Reply:
I guess bump the number to 15.

Reply:
It's not guaranteed the char at 15 is a null...